naga/src/front/wgsl/parse/directive.rs

```diff
@@ -0,0 +1,179 @@
+//! WGSL directives. The focal point of this API is [`DirectiveKind`].
+//!
+//! See also <https://www.w3.org/TR/WGSL/#directives>.
+
+/// A parsed sentinel word indicating the type of directive to be parsed next.
+#[derive(Clone, Copy, Debug, Hash, Eq, PartialEq)]
+pub enum DirectiveKind {
+    Unimplemented(UnimplementedDirectiveKind),
```

Comment:
I feel like we can have all 3 variants here even if they are not implemented but I won't block on this given that we will implement all soon enough.

Reply:
I actually think the `Unimplemented*` type state has been helpful for avoiding extraneous `match`ing in other logic, specifically for the diagnostics. Do you think it's not carrying its weight? 🤔